websocket-tcp-proxy/src/main.rs

```diff
@@ -0,0 +1,112 @@
+use axum::{
+    extract::{
+        ws::{Message, WebSocket, WebSocketUpgrade},
+        Path, TypedHeader,
+    },
+    response::IntoResponse,
+    routing::get,
+    Router,
+};
+use std::net::SocketAddr;
+use tokio::{
+    io::{AsyncReadExt, AsyncWriteExt},
+    net::TcpStream,
+};
+use tower_http::trace::{DefaultMakeSpan, TraceLayer};
+
+#[tokio::main]
+async fn main() {
+    tracing_subscriber::fmt::init();
+
+    let app = Router::new().route("/v1/:ip/:port", get(ws_handler)).layer(
+        TraceLayer::new_for_http().make_span_with(DefaultMakeSpan::default().include_headers(true)),
+    );
+
+    let addr = SocketAddr::from(([127, 0, 0, 1], 3001));
+    tracing::debug!("listening on {}", addr);
+    axum::Server::bind(&addr)
+        .serve(app.into_make_service())
+        .await
+        .unwrap();
+}
+
+async fn ws_handler(
+    Path((ip, port)): Path<(String, String)>,
+    ws: WebSocketUpgrade,
+    user_agent: Option<TypedHeader<headers::UserAgent>>,
+) -> impl IntoResponse {
+    println!("ip: {}, port: {}", ip, port);
+    if let Some(TypedHeader(user_agent)) = user_agent {
+        println!("`{}` connected", user_agent.as_str());
```

Comment:
you're using printlns here but tracing::info below, should stick to one style

Reply:
Fixed